drools-ha/ha-core-infra/src/main/java/org/kie/hacep/core/infra/consumer/DefaultKafkaConsumer.java

```diff
@@ -93,28 +96,50 @@ public DefaultKafkaConsumer(EnvConfig config, Producer producer) {
         }
         this.producer = producer;
         this.consumerUtilsCore = new ConsumerUtilsCoreImpl();
+        this.snapshotOnDemandUtils = snapshotOnDemandUtils;
     }
 
-    public void initConsumer(ConsumerHandler consumerHandler) {
-        this.consumerHandler = (DroolsConsumerHandler) consumerHandler;
-        this.snapShooter = (DefaultSessionSnapShooter) InfraFactory.getSnapshooter(envConfig);
+    //For test
+    void setupForTest(EnvConfig config, Producer producer,
+                      ConsumerUtilsCore consumerUtilsCore,
+                      DefaultSessionSnapShooter defaultSessionSnapShooter,
+                      SnapshotOnDemandUtils snapshotOnDemandUtils, DroolsConsumerHandler consumerHandler){
+        this.envConfig = config;
+        this.producer = producer;
+        this.consumerUtilsCore = consumerUtilsCore;
+        this.snapShooter = defaultSessionSnapShooter;
+        this.snapshotOnDemandUtils = snapshotOnDemandUtils;
+        this.consumerHandler = consumerHandler;
+        getOrCreateKafkaConsumer();
+        getOrCreateKafkaSecondaryConsumer();
+    }
+
+    public void getOrCreateKafkaConsumer(){
```

Comment:
A method like `getOrCreate` should:
- prevent multiple initialization
- return the value

Please rename it to `initKafkaConsumer` if there is no check nor value to return :)